receiver/otlpreceiver/internal/trace/otlp.go

```diff
@@ -34,7 +37,9 @@ func (r *Receiver) Export(ctx context.Context, req ptraceotlp.ExportRequest) (pt
 	// We need to ensure that it propagates the receiver name as a tag
 	numSpans := td.SpanCount()
 	if numSpans == 0 {
-		return ptraceotlp.NewExportResponse(), nil
+		resp := ptraceotlp.NewExportResponse()
+		resp.PartialSuccess().SetErrorMessage(noSpans)
+		return resp, nil
```

Comment:
I'm not sure this is correct for the OTLP specification.  The spec says:

`If the server receives an empty request (a request that does not carry any telemetry data) the server SHOULD respond with success.`

- https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#full-success
- https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#full-success-1

Reply:
OK, so this approach is invalid, I'll reframe.

Reply:
Do we still need this PR?

Reply:
I want to reuse the test and fix the impl.

Reply:
What exactly are you trying to test?

Reply:
The issue associated talks about it a bit and I felt I could probably try and take an approach where the partial success response would return an error message that indicates to the user that no spans were found in payload. The better fix might be to apply to fix https://github.com/open-telemetry/opentelemetry-collector/issues/5935 now, given the feedback.